### ISSUE TITLE
autohide_bookmarks_and_main_toolbars.css removes theme and goes transparent

### ISSUE THREAD

**Issue by ceodoe** [MrOtherGuy/firefox-csshacks#106]
Hey there, thanks for some great hacks. I was recommended autohide_bookmarks_and_main_toolbars.css on reddit and it works exactly like I like it too, but it seems to remove my theme and also goes transparent after the css is applied. Is there any quick and dirty fix I could do to get it to follow my theme and go back to being opaque? Thanks a lot! 

Screenshot:
![image](https://user-images.githubusercontent.com/28344495/119810846-bbf3ba80-bee6-11eb-98dc-58337fae73f0.png)

**Comment by MrOtherGuy** [comment]
It isn't actually possible to do this perfectly. I did apply a patch in #e369c1670c2f2f2769936859e87825495fc7eb4b to have some support for theme images. But you can easily find some themes where it doesn't work well. Still, better than nothing I suppose.

I'm still debating whether or not I should make same kind of change to autohide_bookmarks_toolbar.css. But that style requires some more code and the "imageless" area is much smaller so perhaps it's not worth it to complicate it too much for little benefit.

**Comment by ceodoe** [comment]
That's great! It now looks as expected, but a mysterious 3px gap has appeared between tabs and toolbars:

![image](https://user-images.githubusercontent.com/28344495/119907884-93ed7100-bf51-11eb-88d2-271bd9176132.png)
I use the following CSS-files from your repo:

compact_proton.css
non_floating_sharp_tabs.css
tab_separator_lines.css
iconized_main_menu.css
inline_tab_audio_icons.css
autohide_bookmarks_and_main_toolbars.css
compact_urlbar_megabar.css

They are imported into userChrome in that order. I've removed all other userChrome customizations I used to have

**Comment by MrOtherGuy** [comment]
That should be something caused by `non_floating_sharp_tabs.css`. The line is supposed to be using the text color with less opacity - controlled by your theme, but  Firefox 88 doesn't support the method yet so it is rendered black. Besides, it's not surprising if Firefox 88 versions of both Proton with modifications looks off. 

At any rate, if the line is what I think it is, then you should be able to disabled by adding this bit at the end of your userChrome.css: `#navigator-toolbox{ --tabs-border-color: transparent !important; }`

If that doesn't remove it then I have no idea what it is.

**Comment by MrOtherGuy** [comment]
Hmm. Could you share your whole custom css setup, as well as link the theme you are using? I can then try to find what's going on there. 

By using `#navigator-toolbox{ --tabs-border-color: transparent !important; }` you would set that border to be completely invisible (transparent is a color keyword meaning roughly "invisible"). Except, if that doesn't remove that line, then in your case it's not caused by what I thought it would be.

**Comment by ceodoe** [comment]
I cloned the repo into my chrome folder and did imports in userChrome.css:

```
/* Proton tweaks */
@import url("chrome/compact_proton.css");
@import url("chrome/non_floating_sharp_tabs.css");
@import url("chrome/tab_separator_lines.css");

@import url("chrome/iconized_main_menu.css");
@import url("chrome/inline_tab_audio_icons.css");

/* Others */
@import url("chrome/autohide_bookmarks_and_main_toolbars.css");
@import url("chrome/compact_urlbar_megabar.css");
```

I temporarily commented out compact_proton.css since that fixes the problem. I removed all other custom CSS when I discovered your repo so the only contents of my userChrome.css is what's listed above!

**Comment by MrOtherGuy** [comment]
Am I correct in assuming you are modifying Firefox 88? Whatever pre-release version of proton available in 88 is very out of date and thus there's not really any reason to make the styles work in Firefox 88.

Also btw, where did you put that line I mentioned previously `#navigator-toolbox{ --tabs-border-color: transparent !important; }`? A correct place for it would be after the last `@import` statement in userChrome.css

**Comment by ceodoe** [comment]
I'm running 64-bit FF 89.0 - and yes, that line was added to the very end after the imports.

**Comment by MrOtherGuy** [comment]
Turns out this issue does not happen on Windows. But I was able to reproduce it with Linux Fedora, although I'm still kind of unsure what makes Firefox compute the urlbar height differently on Linux vs. Windows. 

Nonetheless, the patch I made removes a style rule that is supposed to decrease padding around urlbar. The change doesn't seem to have any visible effect on Windows but it makes the toolbar correct height on Linux. Unsure how MacOS will behave though. That rule was possibly something that affected some old version of Proton but is nowadays fine to remove.
